js/battle.js

```diff
@@ -3219,7 +3219,28 @@ var Battle = (function () {
 			if (kwargs.prepare || kwargs.anim === 'prepare') {
 				this.prepareMove(pokemon, move, target);
 			} else if (!kwargs.notarget) {
-				(kwargs.anim ? Tools.getMove(kwargs.anim) : move).anim(this, [pokemon.sprite, target.sprite]);
+				var usedMove = kwargs.anim ? Tools.getMove(kwargs.anim) : move;
+				if (kwargs.spread) {
+					var targets = [pokemon.sprite];
+					var hitPokemon = kwargs.spread;
+					var locId;
+					for (var i = 0; i < target.side.active.length; i++) {
```

Comment:
I suppose, though that still mean you have to loop through both sides for every entry and make the location doesn't it. It looks like that would just move the iteration from `indexOf` to here.